core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -104,6 +105,7 @@ static void init() {
           ConfigurationUtils.searchPropertiesFile(Constants.SITE_PROPERTIES, confPathList);
       // Update site properties and system properties in order
       if (siteProps != null) {
+        discardIgnoredSiteProperties(siteProps);
```

Comment:
shoudn't this be between `merge(siteProps)` and `merge(systemProps)`?

Reply:
No, we first filter out the site-properties to be ignored, then merge the rest.